### PR TITLE
feat(es/minifier): Inline arrow with call expr

### LINKED ISSUE #11007: `es/minifier`: Able to inline function when called nested
### Input Code
```js
const profile = (_s, fn) => {
  return fn();
};

profile("trace1", () => {
  someFunction({
    args1: profile("trace2", () => JSON.stringify(someObj)), // if this line removed, the issue will not reproduce
    args2: JSON.stringify(someObj),
  })
});
```

### Desired Output
```js
someFunction({
    args1: JSON.stringify(someObj),
    args2: JSON.stringify(someObj)
});
```

### Actual Output
```js
const s = (s, r)=>r();
s("trace1", ()=>{
    someFunction({
        args1: s("trace2", ()=>JSON.stringify(someObj)),
        args2: JSON.stringify(someObj)
    });
});
```

#### Summary
Currently, swc does not fully inline and minify code when there are nested calls to a function. It appears that the nested invocation prevents successful inlining, so the minified output still contains the wrapper function. Ideally, the minifier should be able to inline these nested calls when their result is statically determinable and remove the wrappers, resulting in cleaner output.

### Babel plugin or link to the feature description

_No response_

### Additional context

It seems that the bug or limitation is caused by nested calls to the profiling function. If the inner `profile` call is removed, inlining works as expected. Please investigate improvements to the minifier so that it can handle such nested patterns.

### PR DESCRIPTION
**Pull request by Austaras**
**Description:**

**BREAKING CHANGE:**

**Related issue (if exists):**
Closes #11007

### CODE REVIEW

File: crates/swc_ecma_minifier/src/compress/optimize/inline.rs [hunk 1]
```diff
@@ -990,6 +990,16 @@ fn is_arrow_body_simple_enough_for_copy(e: &Expr) -> Option<u8> {
...
+        Expr::Call(c) => {
+            let mut arg_cost = 0;
+            for arg in &c.args {
+                let cost = is_arrow_body_simple_enough_for_copy(&arg.expr)?;
+                arg_cost += cost;
+                arg_cost += if arg.spread.is_some() { 3 } else { 0 };
+            }
+
+            return Some(arg_cost + 2);
+        }
```
Comment (Austaras) [inline crates/swc_ecma_minifier/src/compress/optimize/inline.rs:1004]: This is a very rough estimation, it might cause some problems later -- but at this point perf improves.

**Comment by kdy1** [comment]
@claude Add tests for edge cases

**Comment by kdy1** [comment]
@Austaras Can you run `git fetch --all; git cherry-pick da7d3ecfc`?

**Comment by Austaras** [comment]
I'd rather not. All 3 add cases are edge cases at the call site, but potential problems should raise in the body of arrow function.

File: crates/swc_ecma_minifier/src/compress/optimize/inline.rs [hunk 2]
```diff
@@ -990,6 +990,16 @@ fn is_arrow_body_simple_enough_for_copy(e: &Expr) -> Option<u8> {
...
         }
+        Expr::Call(c) => {
+            let mut arg_cost = 0;
+            for arg in &c.args {
+                let cost = is_arrow_body_simple_enough_for_copy(&arg.expr)?;
+                arg_cost += cost;
+                arg_cost += if arg.spread.is_some() { 3 } else { 0 };
+            }
+
+            return Some(arg_cost + 2);
```
[the remark this answers is not in the document]
Comment (kdy1) [reply crates/swc_ecma_minifier/src/compress/optimize/inline.rs]: @Austaras Do you think this makes sense? For me it sounds reasonable (at the first glance)

**Review by kdy1 — APPROVED** [review]
